lib/jit_preloader.rb

```diff
@@ -22,6 +22,7 @@
 require 'jit_preloader/preloader'
 
 module JitPreloader
+  Warning[:deprecated] = true
```

Comment:
This looks like it will set `Warning[:deprecated]` true when this file gets parsed.  Is that what we want?  Shouldn't we be isolating this to the specs for this gem, putting that line in the spec_helper.rb file?

Reply:
The point of this PR is to enable deprecated warnings with CI run so they can be fixed. However, the PR does not intend to be merged, so I think it matters little where I put it as long as it enables the warnings properly.